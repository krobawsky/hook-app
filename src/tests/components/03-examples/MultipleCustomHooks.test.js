import React from 'react';
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';

jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');

describe('Pruebas en <MultipleCustomHooks />', () => {

    test('debe mostrarse correctamente', () => {
        
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        useCounter.mockReturnValue({
            counter: 10,
            increment: () => {}
        });
    
        const wrapper = shallow(<MultipleCustomHooks />);
        expect( wrapper ).toMatchSnapshot();

    })

    test('debe de mostar la informacion', () => {
 
        useFetch.mockReturnValue({
            data: [{
                author: 'Ricardo',
                quote: 'Hi'
            }],
            loading: false,
            error: null
        });

        useCounter.mockReturnValue({
            counter: 1,
            increment: () => {}
        });
    
        const wrapper = shallow(<MultipleCustomHooks />);
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('.alert').exists() ).toBe(false);
        expect( wrapper.find('.mb-0').text().trim() ).toBe('Hi');
        expect( wrapper.find('footer').text().trim() ).toBe('Ricardo');
        
    })

})