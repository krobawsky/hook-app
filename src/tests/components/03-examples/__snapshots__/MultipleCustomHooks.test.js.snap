// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostar la informacion 1`] = `
<div>
  <h1>
    Braking Bad
  </h1>
  <hr />
  <blockquote
    className="blockquote text-right"
  >
    <p
      className="mb-0 text-right"
    >
      Hi
    </p>
    <footer
      className="blockquote-footer"
    >
      Ricardo
    </footer>
  </blockquote>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;

exports[`Pruebas en <MultipleCustomHooks /> debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Braking Bad
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente
  </button>
</div>
`;
